Fix tests

yurabakhtin opened the pull request:
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix

**The PR fulfills these requirements:**

- [ ] All tests are passing
- [ ] Changelog was modified

**Other information:**
Try to fix errors in the last `master` based PRs https://github.com/humhub/humhub/pull/7486, https://github.com/humhub/humhub/pull/7487.

yurabakhtin commented:
@luke- I have done the commit https://github.com/humhub/humhub/pull/7488/commits/773cc0c8e77d100f7366f2dea4c8e7ec1adb2ed9 as attempt to fix the issue:

![test-artifacts](https://github.com/user-attachments/assets/d0c66dd9-f048-4e03-b5a8-bdafd860544f)

luke- commented:
@yurabakhtin Hmm, problem seems to be that the artifacts are created in different module codeception directories.

Maybe we can improve the upload handling so that either there is only one shared artifact directory via codeception or we get the action to upload `procted/humhub/modules/*/tests/codeception/_output/*`. Maybe we have to specify all modules test dirs manually?

While at it, we should also try uploading other log files (PHP, HumHub) directly? See also here: https://github.com/humhub/humhub/issues/7131#issuecomment-2787402619

yurabakhtin commented:
@luke- After the fix https://github.com/humhub/humhub/pull/7488/commits/773cc0c8e77d100f7366f2dea4c8e7ec1adb2ed9 I can download a zip file with artifacts from all modules [here](https://github.com/humhub/humhub/actions/runs/14351658183)

![fixed-artifacts](https://github.com/user-attachments/assets/0bc2d79e-29c8-42de-813d-8afa1f85eeae)

yurabakhtin commented:
> While at it, we should also try uploading other log files (PHP, HumHub) directly?

@luke- I have added the line `protected/runtime/logs/*`, it helps to read the `app.log`, and I have done several changes to fix errors from the log file, however I could not reproduce the errors and don't have idea, for example, the most unclear error is:

```
2025-04-09 15:42:56 [127.0.0.1][1][-][error][yii\base\Exception] yii\base\Exception: Could not get contents underlying object! - contentid: 9 in /home/runner/work/humhub/humhub/protected/humhub/modules/stream/actions/Stream.php:409
Stack trace:
#0 /home/runner/work/humhub/humhub/protected/humhub/modules/stream/actions/Stream.php(371): humhub\modules\stream\actions\Stream->getStreamEntryResult()
#1 /home/runner/work/humhub/humhub/protected/humhub/modules/stream/actions/Stream.php(350): humhub\modules\stream\actions\Stream->addResponseEntries()
#2 [internal function]: humhub\modules\stream\actions\Stream->run()
#3 /home/runner/work/humhub/humhub/protected/vendor/yiisoft/yii2/base/Action.php(93): call_user_func_array()
#4 /home/runner/work/humhub/humhub/protected/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\Action->runWithParams()
#5 /home/runner/work/humhub/humhub/protected/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction()
#6 /home/runner/work/humhub/humhub/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\base\Module->runAction()
#7 /home/runner/work/humhub/humhub/protected/vendor/yiisoft/yii2/base/Application.php(384): yii\web\Application->handleRequest()
#8 /home/runner/work/humhub/humhub/index-test.php(49): yii\base\Application->run()
#9 {main}
```

I see the error for Content with ID = 2 and 9, however in DB dump I see the content record has correct existing model `humhub\modules\post\models\Post`.
It seems the wrong Content breaks the wall stream.

---

Also I have added the script in order to upload DB dump:

```
            - name: Dump Database
              if: failure()
              run: |
                mkdir -p dump
                mysqldump -u root -proot -h 127.0.0.1 humhub_test > dump/db-dump.sql
```

But it seems the dump is created only at the end, i.e. we cannot know what DB was after each error.

yurabakhtin commented:
@luke- 

> But it seems the dump is created only at the end, i.e. we cannot know what DB was after each error.

I have replaced the "end/total" dump with saving dump after each acceptance failed test - commit https://github.com/humhub/humhub/pull/7488/commits/18587109fc9bb0960930a1a89c28e077f499be37
